assets/js/protocol/navigation.js

```diff
@@ -169,23 +182,65 @@ MzpNavigation.onClick = (e) => {
 };
 
 /**
- * Set initial ARIA navigation states.
+ * use Intersection Observer API to keep track of when the mobile
+ * nav menu is displayed to handle aria roles better
  */
-MzpNavigation.setAria = () => {
-    for (let i = 0; i < _navItemsLists.length; i++) {
-        _navItemsLists[i].setAttribute('aria-expanded', false);
+(MzpNavigation.menuButtonVisible = (callback) => {
+    // check if Intersection observer is supported
+    if (MzpSupports !== 'undefined' && MzpSupports.intersectionObserver) {
+        const observer = new IntersectionObserver(
+            (entries) => {
+                entries.forEach((entry) => {
+                    callback(entry.intersectionRatio > 0, entry.target);
+                });
+            },
+            { root: document.documentElement }
+        );
+        _navMenuButtons.forEach((button) => {
+            observer.observe(button);
+        });
```

Comment:
Suggestion: there's still a `forEach` here on a node list here. It might be ok as most browsers that support Intersection Observer also seem to support `forEach`? But it also might be easier to stick to the same method throughout this file to avoid accidental future error?